serf/serf_test.go

```diff
@@ -174,6 +174,9 @@ func TestSerf_eventsLeave(t *testing.T) {
 	eventCh := make(chan Event, 4)
 	s1Config := testConfig()
 	s1Config.EventCh = eventCh
+	// Make the reap interval longer in this test
+	// so that the leave does not also cause a reap
+	s1Config.ReapInterval = 30 * time.Second
```

Comment:
Was this hiding the issue with the leave not being propagated in this test?

Reply:
Not exactly,  default reap interval in the test is too small, and the leave now waits before the method returns so it was causing this test to fail because the left node also got reaped.